arduino-ide-extension/src/common/protocol/boards-service.ts

```diff
@@ -647,3 +648,127 @@ export function sanitizeFqbn(fqbn: string | undefined): string | undefined {
   const [vendor, arch, id] = fqbn.split(':');
   return `${vendor}:${arch}:${id}`;
 }
+
+export interface BoardConfig {
+  selectedBoard?: Board;
+  selectedPort?: Port;
+}
+
+export interface BoardInfo {
+  /**
+   * Board name. Could be `'Unknown board`'.
+   */
+  BN: string;
+  /**
+   * Vendor ID.
+   */
+  VID: string;
+  /**
+   * Product ID.
+   */
+  PID: string;
+  /**
+   * Serial number.
+   */
+  SN: string;
+}
+
+export const selectBoardForInfo = nls.localize(
+  'arduino/board/selectBoardForInfo',
+  'Please select a board to obtain board info.'
+);
+export function platformMissing(
+  name: string
+): string | BoardInfo | PromiseLike<string | BoardInfo> {
+  return nls.localize(
+    'arduino/board/platformMissing',
+    "The platform for the selected '{0}' board is not installed.",
+    name
+  );
+}
+export const selectPortForInfo = nls.localize(
+  'arduino/board/selectPortForInfo',
+  'Please select a port to obtain board info.'
+);
+export const nonSerialPort = nls.localize(
+  'arduino/board/nonSerialPort',
+  "Non-serial port, can't obtain info."
+);
+export const noNativeSerialPort = nls.localize(
+  'arduino/board/noNativeSerialPort',
+  "Native serial port, can't obtain info."
+);
+export const unknownBoard = nls.localize(
+  'arduino/board/unknownBoard',
+  'Unknown board'
+);
+
+/**
+ * The returned promise resolves to a `BoardInfo` if available to show in the UI or an info message explaining why showing the board info is not possible.
+ */
+export async function getBoardInfo(
+  boardsConfig: BoardConfig,
+  availablePorts: MaybePromise<AvailablePorts>
+): Promise<BoardInfo | string> {
+  const { selectedBoard, selectedPort } = boardsConfig;
+  if (!selectedBoard) {
+    return selectBoardForInfo;
+  }
+  if (!selectedBoard.fqbn) {
+    return platformMissing(selectedBoard.name);
+  }
```

Comment:
I have [the same doubt](https://github.com/arduino/arduino-ide/pull/1803#pullrequestreview-1252844000) about this one. It seems to me that the only thing required is a port selection.